exoscale/api/storage.py

```diff
@@ -343,6 +343,21 @@ def metadata(self):
 
         return res["Metadata"]
 
+    @property
+    def url(self):
+        """
+        Stored file URL.
+
+        Returns:
+            str: the stored file URL
+
+        Note:
+            This property value is dynamically retrieved from the API, incurring extra
```

Comment:
Are you sure about that ? It seems to be only a join between the bucket url and the path.

Reply:
It's not *directly* making an API call but it concatenates the file path to the bucket `zone` property, [which is actually making an API call to get its value](https://github.com/exoscale/python-exoscale/blob/master/exoscale/api/storage.py#L457-L475) so in practice there is an API call being made :)